Wind offshore capacity factor rule fails

### What happened?

The wind offshore capacity factor rule fails, at all spatial resolutions.

### Version

1.2.0.dev

### Relevant log output

```shell
snakemake --profile profiles/conda --configfile="config/default.yaml" "build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv"
Config file config/default.yaml is extended by additional config specified via the command line.
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Job stats:
job                          count    min threads    max threads
-------------------------  -------  -------------  -------------
capacity_factors_offshore        1              1              1
total                            1              1              1

Select jobs to execute...

[Tue Apr  9 11:54:22 2024]
Job 0: Generate capacityfactor time series disaggregated by location on national resolution for wind-offshore.
Reason: Missing output files: build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv

Activating conda environment: ../envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_
Activating conda environment: ../envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_
/Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_/lib/python3.9/site-packages/geopandas/array.py:275: ShapelyDeprecationWarning: The array interface is deprecated and will no longer work in Shapely 2.0. Convert the '.coords' to a numpy array instead.
  return GeometryArray(vectorized.points_from_xy(x, y, z), crs=crs)
Traceback (most recent call last):
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmpwp_m_t5h.capacityfactors_offshore.py", line 84, in <module>
    capacityfactors(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmpwp_m_t5h.capacityfactors_offshore.py", line 55, in capacityfactors
    capacityfactors_per_eez = area_weighted_time_series(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 33, in area_weighted_time_series
    data={
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 34, in <dictcomp>
    shape_id: weighted_time_series(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/lib/eurocalliopelib/geo/spatiotemporal.py", line 84, in weighted_time_series
    assert ds.weight.sum() >= gridcell_overlap_threshold, ds.weight.sum()
AssertionError: <xarray.DataArray 'weight' ()>
array(0.58280955)
Coordinates:
    shape_id  int64 64459
[Tue Apr  9 11:55:28 2024]
Error in rule capacity_factors_offshore:
    jobid: 0
    input: build/data/eez.geojson, build/data/national/shared-coast.csv, data/automatic/capacityfactors/wind-offshore-timeseries.nc
    output: build/models/national/timeseries/supply/capacityfactors-wind-offshore.csv
    conda-env: /Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/9498f8bccfbf440ed1c00845fc76b38a_

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-04-09T115421.386948.snakemake.log
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -56,6 +56,7 @@
 ### Fixed (workflow)
 
 * **FIX** fixed optimisation tolerance of hydro power plants from xtol to xatol (#266).
+* **FIX** source of Exclusive Economic Zones (EEZ) to use a cache on [euro-calliope-datasets](https://github.com/calliope-project/euro-calliope-datasets) so that we can keep using v11 (#332).
 
 ## 1.1.0 (2021-12-22)
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -56,7 +56,7 @@
 ### Fixed (workflow)
 
 * **FIX** fixed optimisation tolerance of hydro power plants from xtol to xatol (#266).
-* **FIX** source of Exclusive Economic Zones (EEZ) to use a cache on [zenodo](https://sandbox.zenodo.org/records/45135) so that we can keep using v11 (#332).
+* **FIX** source of Exclusive Economic Zones (EEZ) to use a cache on [zenodo](https://sandbox.zenodo.org/records/45135) so that we can keep using v11 (#332).  FIXME: update to actual zenodo record before next Euro-Calliope release.
 * **FIX** fixed rule `download_basins_database`, which previously failed on some linux and mac machines, by requiring a more recent curl in the environment `envs/shell.yaml` (#267).
 
 ## 1.1.0 (2021-12-22)

--- a/config/default.yaml
+++ b/config/default.yaml
@@ -1,7 +1,7 @@
 # For descriptions of the configuration values, see `./schema.yaml`.
 data-sources:
     biofuel-potentials-and-costs: https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/ENSPRESO/ENSPRESO_BIOMASS.xlsx
-    eez: https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/add-eez/eez/eez.gpkg.zip # FIXME move to Zenodo?
+    eez: https://sandbox.zenodo.org/records/45135/files/eez_v11.gpkg.zip?download=1
     hydro-generation: https://zenodo.org/record/5797549/files/hydro-generation.csv?download=1
     national-phs-storage-capacities: https://zenodo.org/record/5797549/files/pumped-hydro-storage-capacities-gwh.csv?download=1
     capacity-factors: https://zenodo.org/record/3899687/files/{filename}?download=1

--- a/config/default.yaml
+++ b/config/default.yaml
@@ -1,7 +1,7 @@
 # For descriptions of the configuration values, see `./schema.yaml`.
 data-sources:
     biofuel-potentials-and-costs: https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/ENSPRESO/ENSPRESO_BIOMASS.xlsx
-    eez: https://sandbox.zenodo.org/records/45135/files/eez_v11.gpkg.zip?download=1
+    eez: https://sandbox.zenodo.org/records/45135/files/eez_v11.gpkg.zip?download=1  # FIXME: update to actual zenodo record before next Euro-Calliope release.
     hydro-generation: https://zenodo.org/record/5797549/files/hydro-generation.csv?download=1
     national-phs-storage-capacities: https://zenodo.org/record/5797549/files/pumped-hydro-storage-capacities-gwh.csv?download=1
     capacity-factors: https://zenodo.org/record/3899687/files/{filename}?download=1
@@ -14,8 +14,8 @@ data-sources:
     entsoe-tyndp: https://2020.entsos-tyndp-scenarios.eu/wp-content/uploads/2020/06/TYNDP-2020-Scenario-Datafile.xlsx.zip
     jrc-ppdb: https://zenodo.org/record/3574566/files/JRC-PPDB-OPEN.ver1.0.zip
     jrc-idees: https://jeodpp.jrc.ec.europa.eu/ftp/jrc-opendata/JRC-IDEES/JRC-IDEES-2015_v1/JRC-IDEES-2015_All_xlsx_{country_code}.zip
-    eurostat-energy-balance: https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/feature-sector-coupling/eurostat/nrg_bal_c.tsv.gz # FIXME do not use cached data
-    eurostat-hh-end-use:  https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/feature-sector-coupling/eurostat/nrg_d_hhq.tsv.gz # FIXME do not use cached data
+    eurostat-energy-balance: https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/feature-sector-coupling/eurostat/nrg_bal_c.tsv.gz # FIXME: move to zenodo
+    eurostat-hh-end-use:  https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/feature-sector-coupling/eurostat/nrg_d_hhq.tsv.gz # FIXME: move to zenodo
     swiss-end-use: https://www.bfe.admin.ch/bfe/en/home/versorgung/statistik-und-geodaten/energiestatistiken/energieverbrauch-nach-verwendungszweck.exturl.html/aHR0cHM6Ly9wdWJkYi5iZmUuYWRtaW4uY2gvZGUvcHVibGljYX/Rpb24vZG93bmxvYWQvOTg1NA==.html
     swiss-energy-balance: https://www.bfe.admin.ch/bfe/en/home/versorgung/statistik-und-geodaten/energiestatistiken/gesamtenergiestatistik.exturl.html/aHR0cHM6Ly9wdWJkYi5iZmUuYWRtaW4uY2gvZGUvcHVibGljYX/Rpb24vZG93bmxvYWQvNzUxOQ==.html
     swiss-industry-energy-balance: https://www.bfe.admin.ch/bfe/en/home/versorgung/statistik-und-geodaten/energiestatistiken/teilstatistiken.exturl.html/aHR0cHM6Ly9wdWJkYi5iZmUuYWRtaW4uY2gvZGUvcHVibGljYX/Rpb24vZG93bmxvYWQvODc4OA==.html

--- a/config/default.yaml
+++ b/config/default.yaml
@@ -1,7 +1,7 @@
 # For descriptions of the configuration values, see `./schema.yaml`.
 data-sources:
     biofuel-potentials-and-costs: https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/ENSPRESO/ENSPRESO_BIOMASS.xlsx
-    eez: https://geo.vliz.be/geoserver/MarineRegions/wfs?service=WFS&version=1.0.0&request=GetFeature&typeNames=MarineRegions:eez&outputFormat=SHAPE-ZIP
+    eez: https://raw.githubusercontent.com/calliope-project/euro-calliope-datasets/add-eez/eez/eez.gpkg.zip # FIXME move to Zenodo?
     hydro-generation: https://zenodo.org/record/5797549/files/hydro-generation.csv?download=1
     national-phs-storage-capacities: https://zenodo.org/record/5797549/files/pumped-hydro-storage-capacities-gwh.csv?download=1
     capacity-factors: https://zenodo.org/record/3899687/files/{filename}?download=1

--- a/rules/shapes.smk
+++ b/rules/shapes.smk
@@ -121,7 +121,7 @@ rule eez:
     shell:
         """
         fio cat --bbox {params.bounds} "zip://{input}"\
-        | fio filter "f.properties.territory1 in [{params.countries}]"\
+        | fio filter "f.properties.TERRITORY1 in [{params.countries}]"\
         | fio collect > {output}
         """
 

--- a/rules/shapes.smk
+++ b/rules/shapes.smk
@@ -103,7 +103,7 @@ rule units_without_shape:
 
 rule download_eez:
     message: "Download Exclusive Economic Zones as zip"
-    output: protected("data/automatic/eez.zip")
+    output: protected("data/automatic/eez.gpkg.zip")
     params: url = config["data-sources"]["eez"]
     conda: "../envs/shell.yaml"
     shell: "curl -sLo {output} '{params.url}'"

--- a/scripts/wind-and-solar/capacityfactors_offshore.py
+++ b/scripts/wind-and-solar/capacityfactors_offshore.py
@@ -22,7 +22,7 @@ def capacityfactors(
     final_year=None,
 ):
     """Generate offshore capacityfactor time series for each location."""
-    eez = gpd.read_file(path_to_eez).set_index("mrgid").to_crs(EPSG3035).geometry
+    eez = gpd.read_file(path_to_eez).set_index("MRGID").to_crs(EPSG3035).geometry
     shared_coast = pd.read_csv(path_to_shared_coast, index_col=0)
     shared_coast.index = shared_coast.index.map(lambda x: x.replace(".", "-"))
     shared_coast.columns = shared_coast.columns.astype(int)
@@ -83,11 +83,11 @@ if __name__ == "__main__":
         path_to_timeseries=snakemake.input.timeseries,
         cf_threshold=float(snakemake.params.cf_threshold),
         gridcell_overlap_threshold=float(snakemake.params.gridcell_overlap_threshold),
-        first_year=str(snakemake.params.first_year)
-        if snakemake.params.trim_ts
-        else None,
-        final_year=str(snakemake.params.final_year)
-        if snakemake.params.trim_ts
-        else None,
+        first_year=(
+            str(snakemake.params.first_year) if snakemake.params.trim_ts else None
+        ),
+        final_year=(
+            str(snakemake.params.final_year) if snakemake.params.trim_ts else None
+        ),
         path_to_result=snakemake.output[0],
     )

